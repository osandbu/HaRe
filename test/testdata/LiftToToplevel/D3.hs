module LiftToToplevel.D3 where

{-lift 'sq' to top level. This refactoring only affects the 
  current module. -}
 
sumSquares (x:xs) = sq x + sumSquares xs
  where 
     sq x = x ^ pow

sumSquares [] = 0

pow = 2

