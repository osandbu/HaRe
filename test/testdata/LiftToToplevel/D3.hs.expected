module LiftToToplevel.D3 where

{-lift 'sq' to top level. This refactoring only affects the 
  current module. -}

sumSquares (x:xs) = sq x + sumSquares xs



sumSquares [] = 0

sq x = x ^ pow


pow = 2

